OpenEdXMobile/gradle_scripts/AndroidHelper.gradle

```diff
@@ -99,6 +99,65 @@ class AndroidHelper {
         }
         return available
     }
+
+    static def saveMicrosoftAuthConfigs(project, microsoftConfig) {
+        def microsoftConfigsJsonPath = project.projectDir.toString() + "/res/raw/auth_config.json"
+        new FileWriter(microsoftConfigsJsonPath).withWriter {
+            getMicrosoftConfigurations(project, microsoftConfig).writeTo(it)
+        }
+    }
+
+    static def getMicrosoftConfigurations(project, microsoftConfig) {
+        if (isValidConfigurations(microsoftConfig)) {
+            def jsonBuilder = new JsonBuilder()
+            def clientId = microsoftConfig.get("CLIENT_ID")
+            def signatureHash = microsoftConfig.get("SIGNATURE_HASH")
+            def audienceType = microsoftConfig.get("AUDIENCE_TYPE")
+            def tenantId = microsoftConfig.get("TENANT_ID")
+            jsonBuilder {
+                "client_id" "$clientId"
+                "authorization_user_agent" "DEFAULT"
+                "redirect_uri" "msauth://" + project.APPLICATION_ID + "/" + java.net.URLEncoder.encode(signatureHash, "UTF-8")
```

Comment:
java.net.URLEncoder is an unnecessary qualified reference, you can just use `URLEncoder.encode` instead.

Reply:
Done.